File: backend/models/Sentence.js
const mongoose = require('mongoose');

const sentenceSchema = new mongoose.Schema({
    theme: Number,
    sentence: String,
    blankWord: String,
    correctSentence: String,
    choices: [String],
    koreanMeaning: String
});

const Sentence = mongoose.model('Sentence', sentenceSchema);
module.exports = Sentence;

File: hanaso_front/lib/model/sentence.dart
class Sentence {
  final int theme;
  final String sentence;
  final String blankWord;
  final String koreanMeaning;
  final List<String> choices;

  Sentence({
    required this.theme,
    required this.sentence,
    required this.blankWord,
    required this.koreanMeaning,
    required this.choices,
  });

  factory Sentence.fromJson(Map<String, dynamic> json) {

    return Sentence(
      theme: json['theme'],
      sentence: json['sentence'],
      blankWord: json['blankWord'],
      koreanMeaning: json['koreanMeaning'],
      choices: List<String>.from(json['choices'].map((x) => x)),
    );
  }

}